internal/core/src/segcore/ConcurrentVector.h

```diff
@@ -541,4 +541,17 @@ class ConcurrentVector<BFloat16Vector>
     }
 };
 
+static bool
+ConcurrentDenseVectorCheck(const VectorBase* vec_base, DataType data_type) {
```

Comment:
Is it possible that even if dynamic_cast succeeds, the original pointer is still nullptr?